Detections/MultipleDataSources/UnusualGuestActivity.yaml

```diff
@@ -0,0 +1,51 @@
+id: acc4c247-aaf7-494b-b5da-17f18863878a
+name: Unusual guest activity - Inviting external guest user followed by Azure AD powershell signin activity
+description: |
+  'By default guests have capability to invite more external guest user, who can do suspicious Azure AD enumeration. This detection will first look at guests 
+  inviting external guests users who are then logging via Azure AD powershell after accpeting invitation.
+  Ref : 'https://danielchronlund.com/2021/11/18/scary-azure-ad-tenant-enumeration-using-regular-b2b-guest-accounts/'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - Persistence
+  - Discovery
+relevantTechniques:
+  - T1078.004
+  - T1136.003
+  - T1087.004
+query: |
+  AuditLogs 
+  | where OperationName in ("Invite external user", "Bulk invite users - started (bulk)","Invite external user with reset invitation status")
+  | extend InitiatedByUser = iff(isnotempty(tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName)), 
+    tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName), tostring(parse_json(tostring(InitiatedBy.app)).displayName))
+  | where  InitiatedByUser has_any ("live.com#", "#EXT#")
+  | extend parsedUser = iff(InitiatedByUser has "live.com#", tostring(split(InitiatedByUser, "#")[1]),tostring(split(InitiatedByUser, "#EXT#")[1])) ,  InvitationTime = TimeGenerated
+  | join ( 
+  SigninLogs 
+  | where UserType == "Guest" and AppDisplayName == "Microsoft Azure PowerShell"
+  | extend SigninTime = TimeGenerated
+  ) on $left.parsedUser == $right.UserPrincipalName
+  | project InvitationTime, SigninTime, InitiatedByUser, OperationName, AppDisplayName , IPAddress, UserType
+  | extend timestamp = InvitationTime, AccountCustomEntity = InitiatedByUser, IPCustomEntity = IPAddress

```

Comment:
I do not think we need to include these any longer.

Reply:
removed columns and mapped to the fields directly